.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,354 @@
+        with:
+          fetch-depth: 0
+
+      - name: Get previous git tag
+        id: previous-tag
+        run: |
+          tag=$(git for-each-ref --sort=-creatordate --count 5 --format="%(refname:short)" "refs/tags/" | grep -E "^[0-9]+\.[0-9]+\.[0-9]+" | head -n 1)
+          echo "Found tag: $tag"
+          echo "tag=$(echo tag)" >> $GITHUB_OUTPUT
+
+      - name: Get next semver from previous tag
+        id: next-semvers
+        uses: WyriHaximus/github-action-next-semvers@v1
+        with:
+          version: ${{ steps.previous-tag.outputs.tag }}
+
+      - name: Set PR version
+        if: needs.set-metadata.outputs.build-type == 'PR'
+        id: set-pr-version
+        run: |
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          PR_ITERATION=${{ github.run_attempt }}.${{ github.run_number }}
+          echo "version=$(echo ${{ steps.next-semvers.outputs.patch }}-alpha+PR.${PR_NUMBER}.${PR_ITERATION})" >> $GITHUB_OUTPUT
+
+      - name: Set Branch version
+        if: needs.set-metadata.outputs.build-type == 'branch'
+        id: set-branch-version
+        run: |
+          BRANCH_ITERATION=${{ github.run_attempt }}.${{ github.run_number }}
+          echo "version=$(echo ${{ steps.next-semvers.outputs.patch }}-alpha+BR.${{ github.ref_name }}.${BRANCH_ITERATION})" >> $GITHUB_OUTPUT
+
+      - name: Make Version PEP-Compliant
+        if: steps.set-pr-version != '' || steps.set-branch-version != ''
+        id: pepify
+        shell: bash
+        run: |
+          if [[ -n "${{ steps.set-pr-version.outputs.version }}" ]]; then
+            echo "Found PR version to pepify..."
+            semver=${{ steps.set-pr-version.outputs.version }}
+          else
+            echo "Found branch version to pepify..."
+            semver=${{ steps.set-branch-version.outputs.version }}
+          fi
+          # Per PEP 440, the full scope of traditional semantic versioning is not valid in Python.
+          int_only_br=$(echo "${{ github.ref_name }}" | sed 's/[^0-9]*//g')
+          metadata=".dev$int_only_br"
+          pepified_version=$(echo "$semver" | sed -r -e 's/-alpha/a0/g' -e 's/-beta/b0/g' -e 's/\+PR./\.dev/g' -e "s|\+BR.${{ github.ref_name }}.|$metadata|g" -e 's/([0-9]+\.[0-9]+\.[0-9]+)([a-b0-9]+)?([\.dev]+)?(([0-9]+)\.([0-9]+)(\.([0-9]+))?)?/\1\2\3\5\6\8/g')
+          echo "Tagged version converted to PEP 440 standard: $pepified_version"
+          echo "version=$pepified_version" >> $GITHUB_OUTPUT
+
+      - name: Construct release tag
+        if: needs.set-metadata.outputs.build-type == 'release'
+        id: set-release-version
+        run: |
+          RELEASE_TAG=""
+
+          # Check for null tag:
+          if [[ "${{ inputs.optional-release-tag }}" = "0.0.0" || (-z "${{ inputs.optional-release-tag }}" && "${{ steps.previous-tag.outputs.tag }}" = "0.0.0") ]]; then
+            echo "❌ - Null tag (0.0.0) is invalid for a release (hint: repo could contain no tags)"
+            exit 1
+          fi
```

Comment:
I believe this check is now invalid. With the previous actions, it defaulted to "0.0.0" if no tags were found. This logic needs to be updated to whatever your new action does when it can't find tags. And if that action fails, then I would just remove this check.